src/zarr/core/group.py

```diff
@@ -290,6 +292,28 @@ async def delitem(self, key: str) -> None:
         else:
             raise ValueError(f"unexpected zarr_format: {self.metadata.zarr_format}")
 
+    async def get(
+        self, key: str, default: DefaultT | None = None
```

Comment:
I don't really know how to type this properly, but IMO as a user I'd appreciate the ability to know that `.get(key, default)` is still `AsyncArray | AsyncGroup` (or None I guess). So maybe we have `DefaultT` also be bound to `AsyncArray, AsyncGroup`?

That said

```python
from typing import reveal_type

foo : dict[str, int] = {}
reveal_type(foo.get("a", "b"))
```

gives

```
src/zarr/core/metadata/v3.py:555: note: Revealed type is "Union[builtins.int, builtins.str]"
```

so nevermind :/